### PR TITLE
[SPARK-49892][SQL] Assign proper error class for _LEGACY_ERROR_TEMP_1136

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to assign proper error condition for _LEGACY_ERROR_TEMP_1136

### Why are the changes needed?

Currently we don't have proper error condition and SQLSTATE when user try saving interval data type into external storage

### Does this PR introduce _any_ user-facing change?

No, only user-facing error message improved

### How was this patch tested?

Updated the existing tests

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

File: common/utils/src/main/resources/error/error-conditions.json [hunk 1]
```diff
@@ -400,6 +400,12 @@
     ],
     "sqlState" : "58030"
   },
+  "CANNOT_SAVE_INTERVAL_TO_EXTERNAL_STORAGE" : {
```
Comment (MaxGekk) [inline common/utils/src/main/resources/error/error-conditions.json:403]: I wonder cannot you re-use `UNSUPPORTED_DATA_TYPE_FOR_DATASOURCE`?

Comment (itholic) [reply common/utils/src/main/resources/error/error-conditions.json]: Sounds good. Just updated

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala [hunk 2]
```diff
@@ -566,12 +569,15 @@ case class DataSource(
...
+      format: String,
       forbidAnsiIntervals: Boolean): Unit = {
-    dataTypes.foreach(
-      TypeUtils.invokeOnceForInterval(_, forbidAnsiIntervals) {
-      throw QueryCompilationErrors.cannotSaveIntervalIntoExternalStorageError()
-    })
+    dataTypesWithNames.foreach { case (dataType, columnName) =>
+      TypeUtils.invokeOnceForInterval(dataType, forbidAnsiIntervals) {
+      throw QueryCompilationErrors.cannotSaveIntervalIntoExternalStorageError(
+        format, toSQLId(columnName), toSQLType(dataType)
```
Comment (MaxGekk) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala:578]: Better to follow existing convention, and apply quoting inside of `cannotSaveIntervalIntoExternalStorageError`

**Review by MaxGekk — COMMENTED** [review]
LGTM except of a comment.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala (hunk 2 above)
Comment (itholic) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala]: Applied the comment. Thanks!

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @itholic.

**Comment by itholic** [comment]
Thanks @MaxGekk for the review!

**Pull request closed without merging**